src/common/utils.ts

```diff
@@ -521,9 +521,22 @@ export class URIUtils {
         return (result && result.params) || {};
     }
 
-    public static compilePath(pathTemplate: string, params: object): string {
+    public static compilePath(pathTemplate: string, params: object, defaultPathParams: object): string {
+        /* Add required default key, value pairs for the "toPath" function into a cloned params object. */
+        const withDefaultParams = (pathTemplate: string, params: object, defaultPathParams: object): object => {
+            const requiredKeys = [];
+            const retParams = {...params};
+            pathToRegexp(pathTemplate, requiredKeys);
+            for (const { name } of requiredKeys) {
+                if (!Object.hasOwnProperty.call(params, name)) {
```

Comment:
nit: I'd suggest explicitly `.call(retParams, name)` here.

Reply:
Change to `retParams.hasOwnProperty(name)` bacause a retParams object must has the hasOwnProperty method.